core/store/models/eth.go

```diff
@@ -216,6 +223,32 @@ func (f *FunctionSelector) UnmarshalJSON(input []byte) error {
 	return nil
 }
 
+// MarshalJSON returns the JSON encoding of f
+func (f FunctionSelector) MarshalJSON() ([]byte, error) {
+	return json.Marshal([]byte(f.String()))
+}
```

Comment:
Why the conversion to bytes before JSON encoding?

I was pleasantly surprised Golang converts the string to Base64 under the hood for []bytes, but was this intended?  Does the unmarshaling work?  The value is unmarshaled back into a 'string', and not a '[]byte'.

Reply:
Thanks, I removed the cast in 0b966cb5f10be58f625c4b12f9ab357ae86eb7b6.